core/src/services/s3/writer.rs

```diff
@@ -192,11 +194,15 @@ impl oio::Write for S3Writer {
         }
     }
 
-    async fn sink(&mut self, _size: u64, _s: oio::Streamer) -> Result<()> {
-        Err(Error::new(
-            ErrorKind::Unsupported,
-            "Write::sink is not supported",
-        ))
+    async fn sink(&mut self, size: u64, s: oio::Streamer) -> Result<()> {
+        if self.op.content_length().unwrap_or_default() == size {
+            return self.write_oneshot(size, AsyncBody::Stream(s)).await;
```

Comment:
😢I thought you were going to implement a separate sink.

Reply:
Can you explain it a bit more? I didn't understand.

Reply:
Sorry. My misunderstanding.